src/h5web/visualizations/heatmap/utils.test.ts

```diff
@@ -0,0 +1,20 @@
+import { interpolateRdBu } from 'd3-scale-chromatic';
+import { generateCSSLinearGradient } from './utils';
+
+describe('generateCSSLinearGradient', () => {
+  it('should generate a linear gradient from first to last color', () => {
+    // Escape parentheses for literal matching
+    const firstColor = interpolateRdBu(0)
+      .replace('(', '\\(')
+      .replace(')', '\\)');
+    const lastColor = interpolateRdBu(1)
+      .replace('(', '\\(')
+      .replace(')', '\\)');
+    const regexp = RegExp(
+      `linear-gradient\\(to top,${firstColor},.*${lastColor}\\)`
+    );
+    expect(regexp.test(generateCSSLinearGradient(interpolateRdBu, 'top'))).toBe(
+      true
+    );
```

Comment:
https://jestjs.io/docs/en/expect#expectstringmatchingstring--regexp

```
expect(result).toBe(expect.stringMatching(regex))
```

Reply:
That being said, to avoid the complex regex creation, you could just use a simple interpolator and hard code the gradient string returned with it in your test.